src/button.js

```diff
@@ -0,0 +1,42 @@
+// // button.component.js
+// import React from 'react';
+// import styles from './button.css';
+// export const Button = ({ text, onClick, type, disabled }) => (
```

Comment:
Rather than creating a callback function to create your Button component, structure your component into a class, similar to how your Map class is structured. Your method of creating the Button component will cause your lifecycle functions (componentDidUpdate({ mapPosition })) to fail.